pandas/tseries/index.py

```diff
@@ -73,16 +74,20 @@ def f(self):
 
             result = libts.get_start_end_field(values, field, self.freqstr,
                                                month_kw)
-        elif field in ['weekday_name']:
-            result = libts.get_date_name_field(values, field)
-            return self._maybe_mask_results(result)
+            return self._maybe_mask_results(result, convert='float64')
         elif field in ['is_leap_year']:
             # no need to mask NaT
             return libts.get_date_field(values, field)
+
+        # non-boolean accessors -> return Index
+        elif field in ['weekday_name']:
```

Comment:
same as above maybe list this in DatetimeIndex, maybe ``_other_ops = ['weekday_name']`` or something
just to avoid explicity listing these in two places.